src/OrchardCore.Modules/OrchardCore.Html/Handlers/HtmlBodyPartHandler.cs

```diff
@@ -1,34 +1,58 @@
-using System.Linq;
+using System.Text.Encodings.Web;
 using System.Threading.Tasks;
+using Fluid;
 using Microsoft.AspNetCore.Html;
 using OrchardCore.ContentManagement.Handlers;
-using OrchardCore.ContentManagement.Metadata;
 using OrchardCore.ContentManagement.Models;
 using OrchardCore.Html.Models;
-using OrchardCore.Html.Settings;
+using OrchardCore.Html.ViewModels;
+using OrchardCore.Liquid;
 
 namespace OrchardCore.Html.Handlers
 {
     public class HtmlBodyPartHandler : ContentPartHandler<HtmlBodyPart>
     {
-        private readonly IContentDefinitionManager _contentDefinitionManager;
+        private readonly ILiquidTemplateManager _liquidTemplateManager;
 
-        public HtmlBodyPartHandler(IContentDefinitionManager contentDefinitionManager)
+        private HtmlString _bodyAspect;
+
+        public HtmlBodyPartHandler(ILiquidTemplateManager liquidTemplateManager)
         {
-            _contentDefinitionManager = contentDefinitionManager;
+            _liquidTemplateManager = liquidTemplateManager;
         }
 
         public override Task GetContentItemAspectAsync(ContentItemAspectContext context, HtmlBodyPart part)
         {
-            return context.ForAsync<BodyAspect>(bodyAspect =>
+            return context.ForAsync<BodyAspect>(async bodyAspect =>
             {
-                var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(part.ContentItem.ContentType);
-                var contentTypePartDefinition = contentTypeDefinition.Parts.FirstOrDefault(p => p.PartDefinition.Name == nameof(HtmlBodyPart));
-                var settings = contentTypePartDefinition.GetSettings<HtmlBodyPartSettings>();
-                var body = part.Html;
+                if (_bodyAspect != null)
+                {
+                    bodyAspect.Body = _bodyAspect;
+                    return;
+                }
+
+                try
+                {
+                    var model = new HtmlBodyPartViewModel()
+                    {
+                        Html = part.Html,
```

Comment:
Fishy to apply the same property to both `Source` and `Html` ...

Reply:
Yes, `Source` was an already existing property that i don't need anymore, i kept it in case someone already use it in a liquid expression. At this stage, before the liquid evaluation, `Source == Html`.

E.g the markdown view model (also the field as i remember) has a `Markdown` property and an `Html` property. The `LiquidPart` view model has a `Liquid` and an `Html` property. Here the singularity is that the `Html` property already exists. Note: the html field doesn't have a `Source` property.

**So if you want i can just remove this `Source` property**.

Note: Here normally we also don't need to pass the `Html` property that we want to evaluate, but i said to me that maybe it would be useful, so just pass the more complete model as we can.

Reply:
Done, i removed the `Source` property